bigquery_storage/noxfile.py

```diff
@@ -13,113 +15,136 @@
-    session.run('flake8', 'tests')
+    system_test_exists = os.path.exists(system_test_path)
+    system_test_folder_exists = os.path.exists(system_test_folder_path)
+    # Sanity check: only run tests if found.
+    if not system_test_exists and not system_test_folder_exists:
+        session.skip("System tests were not found")
 
+    # Use pre-release gRPC for system tests.
+    session.install("--pre", "grpcio")
 
-@nox.session(python='3.6')
-def lint_setup_py(session):
-    """Verify that setup.py is valid (including RST check)."""
-    session.install('docutils', 'pygments')
-    session.run('python', 'setup.py', 'check', '--restructuredtext',
-                '--strict')
+    # Install all test dependencies, then install this package into the
+    # virtualenv's dist-packages.
+    session.install("mock", "pytest")
+    for local_dep in LOCAL_DEPS:
+        session.install("-e", local_dep)
+    session.install("-e", "../test_utils/")
+    session.install("-e", "pandas")
+    session.install("-e", "fastavro")
+    session.install("-e", ".")
 
+    # Run py.test against the system tests.
+    if system_test_exists:
+        session.run("py.test", "--quiet", system_test_path, *session.posargs)
+    if system_test_folder_exists:
+        session.run("py.test", "--quiet", system_test_folder_path, *session.posargs)
 
-@nox.session(python='3.6')
+
+@nox.session(python="3.7")
 def cover(session):
     """Run the final coverage report.
+
     This outputs the coverage report aggregating coverage from the unit
     test runs (not system test runs), and then erases coverage data.
     """
-    session.install('coverage', 'pytest-cov')
-    session.run('coverage', 'report', '--show-missing', '--fail-under=100')
-    session.run('coverage', 'erase')
+    session.install("coverage", "pytest-cov")
+    session.run("coverage", "report", "--show-missing", "--fail-under=100")
 
-
-@nox.session(python=['2.7', '3.6'])
-def system(session):
-    """Run the system test suite."""
-
-    # Sanity check: Only run system tests if the environment variable is set.
-    if not os.environ.get('GOOGLE_APPLICATION_CREDENTIALS', ''):
-        session.skip('Credentials must be set via environment variable.')
-
-    # Install all test dependencies, then install this package into the
-    # virtualenv's dist-packages.
-    session.install('pytest')
-    session.install('-e', os.path.join('..', 'test_utils'))
-    for local_dep in LOCAL_DEPS:
-        session.install('-e', local_dep)
-    session.install('-e', '.[pandas,fastavro]')
-
-    # Run py.test against the system tests.
-    session.run('py.test', '--quiet', 'tests/system/')
+    session.run("coverage", "erase")
 
 
 @nox.session(python='3.6')
 def docs(session):
     """Build the docs."""
-
     session.install('sphinx', 'sphinx_rtd_theme')
     session.install('-e', '.[pandas,fastavro]')
-
     shutil.rmtree(os.path.join('docs', '_build'), ignore_errors=True)
```

Comment:
Want to do the import in this function instead?

Note: I added the `docs` session as an easy way to rebuild just these docs. I recall that other libraries don't have this, but I think it'd be good to standardize on having one because it's nice to be able to build a subset, but also because it's useful for eventual repo split.